src/parser.rs

```diff
@@ -158,14 +158,22 @@ impl PartialOrd for ParsedValue {
 impl Parser {
     /// Add a single entity value, along with its rank, to the parser
     /// The ranks of the other entity values will not be changed
-    pub fn add_value(&mut self, entity_value: EntityValue, rank: u32) {
+    pub fn add_value(&mut self, entity_value: EntityValue, rank: u32) -> Result<()> {
         // We force add the new resolved value: even if it is already present in the symbol table
         // we duplicate it to allow several raw values to map to it
         let res_value_idx = self
             .resolved_symbol_table
             .add_symbol(entity_value.resolved_value);
 
-        for (_, token) in whitespace_tokenizer(&entity_value.raw_value) {
+        let tokens: Vec<(Range<usize>, String)> = whitespace_tokenizer(&entity_value.raw_value)
+            .into_iter()
+            .collect();
+
+        if tokens.len() == 0 {
+            bail!("Can't add value '{}' because it's empty or contains only whitespaces.", entity_value.raw_value)
+        }
```

Comment:
It would be more efficient to use a boolean flag `is_empty` and set it to true in the loop. It will avoid creating a vector, and will make injection slightly faster.

Reply:
Another detail is that the `resolved_symbol_table` should not be updated in case of invalid input.